hack/local-up-cluster.sh

```diff
@@ -131,11 +131,12 @@ ENABLE_ADMISSION_PLUGINS=${ENABLE_ADMISSION_PLUGINS:-"NamespaceLifecycle,LimitRa
 DISABLE_ADMISSION_PLUGINS=${DISABLE_ADMISSION_PLUGINS:-""}
 ADMISSION_CONTROL_CONFIG_FILE=${ADMISSION_CONTROL_CONFIG_FILE:-""}
 
-# START_MODE can be 'all', 'kubeletonly', 'nokubelet', 'nokubeproxy', or 'nokubelet,nokubeproxy'
+# START_MODE can be 'all', 'control-plane', 'worker', 'kubelet', 'kubeproxy'
+# legacy options: 'kubeletonly', 'nokubelet', 'nokubeproxy', or 'nokubelet,nokubeproxy'
```

Comment:
concerned about this, these don't imply the same thing and we have a lot of existing workflows out there, are we actually planning to drop these?

cc @dims

Reply:
Improving / reducing the need to sudo for control plane components is a useful goal, I don't really see why restructuring the option surface is needed to do that

Reply:
The motivation is to add a new worker mode, which starts kubelet + kube-proxy. But I found it hard to fit this into the original design. Consider if we support something like `kubeletonly,kubeproxyonly`, it can be very error-prone to write all the if statements about START_MODE, because it contains both positive and negative words.

So I decided to restructure it. All modes can be expressed in combination of positive words now. nokubelet/nokubeproxy/kubeletonly are kept just for capability.

Reply:
> we actually planning to drop these?
I don't want to drop these. The existing workflow that uses the old modes should still work after this PR.

Reply:
We should not call them "legacy" if there's no plan to drop them, and I don't think we should add both positive and negative options. A number of these options also seem much less clear, "control-plane" and "worker" is not well-defined in the case of a script meant to bring up the host as a single node.

Reply:
> I don't think we should add both positive and negative options.

Agreed, that's why I call `nokubelet` "legacy". They are still supported now, but not recommended any more. Do you have a better word than "legacy"? Maybe "deprecated"?

After deprecating `nokubelet`, we need a new option to specify components that can run on darwin, which should be `control-plane`. Then naturally we need a new option for the opposite. I choose `worker`. I think they are easy to understand.

> meant to bring up the host as a single node.

I actually managed to use this script to bring up a podman container as worker, connecting to the control-plane running natively on darwin, so that I have a fully working cluster on my MacBook. That is to say, I use this script on two hosts to bring up one cluster. I will edit the PR description to include the details.

Reply:
>  That is to say, I use this script on two hosts to bring up one cluster.

With the kind scripts as well ... while you _can_ combine these and make it multi-node, that's really not what this script is supported for and I'm wary of expanding the scope of this script.

Part of the point of this script is to be minimal, not to compete with tools like kubeadm.

Reply:
> Part of the point of this script is to be minimal, not to compete with tools like kubeadm.

I understand the concern. But the existing `kubeletonly` mode already needs a separate control plane deployment to work. I would argue that I just want to start a kube-proxy with kubelet, to make this mode more usable, and does not expand the complexity too much.

It seems that all the concerns are about the newly added START_MODE. So can we get the sudo part in first? Please also help me take a look at #129120